File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> m;
        vector<int> res;
        for(auto num:nums) {
            m[num] = 1 + m[num];
        }
        vector<vector<int>> freq(nums.size());
        for(auto it=m.begin(); it != m.end(); ++it) {
            freq[it->second - 1].push_back(it->first);
        }
        for(int i=freq.size() - 1; i>=0; i--) {
            for(int n:freq[i]) {
                res.push_back(n);
                if(res.size() == k){
                    return res;
                }
            }
        }
        return res;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        counts = {}
        for n in nums:
            if n not in counts:
                counts[n] = 1
            else:
                counts[n] += 1
        return [x[0] for x in sorted(counts.items(), key=lambda y: y[1],reverse=True)][:k]
        